p2p_tc.c

```diff
@@ -43,6 +46,22 @@ static int p2p_delay_request(struct port *p)
 	return port_delay_request(p);
 }
 
+static bool hsr_ring_eliminate(struct port *p, struct ptp_message *msg)
+{
+	struct ClockIdentity c1, c2;
+
+	if (clock_is_hsr(p->clock)) {
+	    c1 = clock_identity(p->clock);
+	    c2 = msg->header.sourcePortIdentity.clockIdentity;
```

Comment:
> Should we check 'c1' and 'c2' for NULL?

They can't be NULL. For c1, a port needs a clock, otherwise something is horribly wrong with ptp4l. For c2, this would only happen if the packet is malformed, which is handled in `msg_post_recv` by verifying that the PTP message type has the correct packet length.